### ISSUE TITLE
Duration of execution of SRW (and other codes) in Sirepo log/JS console

### ISSUE THREAD

**Issue by mrakitin** [radiasoft/sirepo#255]
To compare execution time of SRW (and potentially other codes like Warp and Elegant) from within Sirepo and in console on the same example we need to have the duration of command execution in both Sirepo server log and in the JS console. That would help much to understand where we waste time during execution of SRW in Sirepo. Preliminary testing by @DmitrySmalyuk shows that sometimes we have 2 to 3 times longer execution time in the web interface than in console with the same input parameters to SRW. My suspicion is that Sirepo spends quite long on sending huge data arrays to the JS plotting scripts. Dmitry will perform tests and the results will be attached here.

**Comment by mrakitin** [comment]
Here is an example of monitoring of the duration (NSLS-II FMX beamline with default settings, `Beamline` page, `Intensity at Sample Report, 67m`):
![duration](https://cloud.githubusercontent.com/assets/13209176/17311268/e9ab383a-5817-11e6-9a01-891932e93e9d.png)
Execution of SRW itself took ~7.45 seconds, but the whole processing took ~4 times more - 28.31 seconds.

**Comment by mrakitin** [comment]
The issue seems to be due to huge data files which are sent over network:
![profile](https://cloud.githubusercontent.com/assets/13209176/17311943/fdccacb8-581c-11e6-9834-14e4dcd0d78e.png)

**Execution of SRW:** 8.31s
**Waiting for it in the client:** 11.18s
**Downloading of 12.5 MB file:** 18.25s

To Oleg (@ochubar): please have a look at the results. The reason we have a lag in Sirepo comparing with the execution of SRW in console is that the large files have to be transferred over network to the JS client before we can see the resulted image.

To Paul (@moellep): can we try to zip/compress these files on the server side and decompress them on the client side?

**Comment by robnagler** [comment]
I ran this on beta. It took about 15 seconds. I don't know the speed
difference between the machines, but I do know that it is only taking a few
seconds for the data to arrive. See the screen below.

One problem may be that our nginx server compresses json whereas
nsls2expdev1 may not do that. However, the out file is only 233KB so it
should be very fast locally. Another problem may be the virtual private
network. My browser is connected over the internet to the beta server,
which is running https, which might slow things down a bit.

When I look at top, sirepo is pegged at 100% for most of the time it takes
to run.

**Comment by mrakitin** [comment]
That's possible that the flask server on nsls2expdev1 does not compress data. Please have a try with this example on alpha - https://alpha.sirepo.com/srw#/beamline/w8BZNBAL?simulation.facility=NSLS-II&application_mode=light-sources&show_item_id=PTMxhqO9. The output is 25 MB, but when I do the calculation from home, it takes about 50 seconds for calculation and ~10 seconds for data download.

Could you please generate new docker container with the latest sirepo, so we could know exact SRW execution time? Thanks.

**Comment by robnagler** [comment]
I ran the example on alpha and it took 7.5s to download. The output is 95MB
uncompressed and 24MB compressed.

I updated beta with the current alpha, and alpha with the latest in github.

**Comment by robnagler** [comment]
I removed the 1st priority label, because optimization hasn't been discussed. We should discuss that in tomorrow's meeting if there is one. I added a new issue (#256) for this specific development task (plot download optimization) instead of tying it to SRW, because it's a more general problem.

**Comment by mrakitin** [comment]
Thanks Rob.

**Comment by DmitrySmalyuk** [comment]
Speed of calculation tests performed to compare the efficiency of calculation on the web and the command line interfaces, and efficiency of calculations on different servers. (Including using the "duration of execution" feature.)

[CalculationSpeedAnalysis.xlsx](https://github.com/radiasoft/sirepo/files/399676/CalculationSpeedAnalysis.xlsx)

**Comment by mrakitin** [comment]
From the table we can see that the download time takes 25-50% of all time from starting the report to the moment the plot is displayed in a browser. #256 will take care of the download time. Closing.
